### DISCUSSION TITLE
Improvement in getRevNumber()

### DISCUSSION THREAD

**Discussion by jasp402** [vasani-arpit/WBOT#216] [Ideas]
Greetings to all. I have been working in my spare time with WBOT. and is analyzing how the chromium download issue works.

specifically the [detectRev.js](https://github.com/vasani-arpit/WBOT/blob/master/src/detectRev.js) file. 

```js
this.getRevNumber = function () {
    return new Promise((resolve) => {
        // fetch('https://omahaproxy.appspot.com/all.json?os=' + chromePlatform, { method: "get" })
        //     .then(res => res.json())
        //     .then(json => {
        //         //console.log(json);
        //         var betaChannel = json[0].versions.find(x => x.channel == "canary").branch_base_position;
        resolve("666595");
        // });
    })

}
```
Commented on lines 21-25. I decided to reactivate it and see what was the problem or the reason why it was deactivated. At first it worked very well for me. but as the days went by I started to have problems. I explain a little to put the discovered in context.

https://omahaproxy.appspot.com/ is used to find the latest Chromium distributions that are compatible with PuppeteerJS. for that reason they are filtered based on the "canary" property However, the problem occurs when these versions are not exactly downloadable distributions. 

https://omahaproxy.appspot.com/all.json?os=win
In my case is Rev. **868734**

https://storage.googleapis.com/chromium-browser-snapshots/Win_x64/868734/chrome-win.zip
![image](https://user-images.githubusercontent.com/8978470/113451547-cce0ed80-93c7-11eb-9b66-28c0c231874f.png)

### Why is this failing?
Basically the developers of Cromium create a new version, this version is published and available for download, however when finding serious bugs, the version is removed from the download part. This generates an error in the `downloadAndStartThings()` function. Specifically in the part of [src/index.js](https://github.com/vasani-arpit/WBOT/blob/267188b115a25b5c655ade2c6579d4da1d94e091/src/index.js#L49) `await browserFetcher.download (revNumber)`

Because although the revision number exists, it is not available for download.

---

### Solution:
After reviewing the above. I found a way to be able to download the recent, stable and permanent versions of cromium. Basically, it is to look directly at the download source and from there take the number corresponding to the revision.

https://www.googleapis.com/storage/v1/b/chromium-browser-snapshots/o?&delimiter=/&prefix=[OS_PLATFORM]/&fields=prefixes

the mechanism is almost identical, you just have to use this URL and make a couple of adjustments in the inside of the function so that it correctly searches for what is required. It is still too early to win, more tests are required, more community support to check on different OS and see the impact on performance but it seems like a good solution.

I also understand that fixing the version is due to problems with PuppeteerJS. to safeguard compatibility and stuff. But I always recommend keeping applications up to date and this is a way to automatically update the kernel

```js
const getRevNumber=()=> {
		let chromePlatform = os;
		switch (os) {
			case "linux":
				chromePlatform = "Linux";
				break;
			case "win32":
				chromePlatform = "Win";
				break;
			case "darwin":
				chromePlatform = "Mac";
				break;
		}
		return new Promise(resolve => {
			fetch(`https://www.googleapis.com/storage/v1/b/chromium-browser-snapshots/o?&delimiter=/&prefix=${chromePlatform}/&fields=prefixes`, {method: "get"})
				.then(res => res.json())
				.then(json => {
					let channel = json.prefixes[0].replace(chromePlatform, '').replace(/\//g, '');
					resolve(channel);
				})
				.catch(e => {
					throw new Error(e);
				});
		});
	}
```

**Comment by vasani-arpit** [comment]
Can you make a PR for this? This was implemented earlier but there was an problem with that implementation.
